source/windows/secure_channel_tls_handler.c

```diff
@@ -1812,20 +1916,199 @@ static struct aws_channel_handler *s_tls_handler_new(
 
     return NULL;
 }
+
+static DWORD getEnabledProtocols(const struct aws_tls_ctx_options *options, bool is_client_mode) {
+    DWORD grbitEnabledProtocols = 0;
+    if (is_client_mode) {
+        switch (options->minimum_tls_version) {
+            case AWS_IO_SSLv3:
+                grbitEnabledProtocols |= SP_PROT_SSL3_CLIENT;
+            case AWS_IO_TLSv1:
+                grbitEnabledProtocols |= SP_PROT_TLS1_0_CLIENT;
+            case AWS_IO_TLSv1_1:
+                grbitEnabledProtocols |= SP_PROT_TLS1_1_CLIENT;
+            case AWS_IO_TLSv1_2:
+#if defined(SP_PROT_TLS1_2_CLIENT)
+                grbitEnabledProtocols |= SP_PROT_TLS1_2_CLIENT;
+#endif
+            case AWS_IO_TLSv1_3:
+#if defined(SP_PROT_TLS1_3_CLIENT)
+                grbitEnabledProtocols |= SP_PROT_TLS1_3_CLIENT;
+#endif
+                break;
+            case AWS_IO_TLS_VER_SYS_DEFAULTS:
+                grbitEnabledProtocols = 0;
+                break;
+        }
+    } else {
+        switch (options->minimum_tls_version) {
+            case AWS_IO_SSLv3:
+                grbitEnabledProtocols |= SP_PROT_SSL3_SERVER;
+            case AWS_IO_TLSv1:
+                grbitEnabledProtocols |= SP_PROT_TLS1_0_SERVER;
+            case AWS_IO_TLSv1_1:
+                grbitEnabledProtocols |= SP_PROT_TLS1_1_SERVER;
+            case AWS_IO_TLSv1_2:
+#if defined(SP_PROT_TLS1_2_SERVER)
+                grbitEnabledProtocols |= SP_PROT_TLS1_2_SERVER;
+#endif
+            case AWS_IO_TLSv1_3:
+#if defined(SP_PROT_TLS1_3_SERVER)
+                grbitEnabledProtocols |= SP_PROT_TLS1_3_SERVER;
+#endif
+                break;
+            case AWS_IO_TLS_VER_SYS_DEFAULTS:
+                grbitEnabledProtocols = 0;
+                break;
+        }
+    }
+    return grbitEnabledProtocols;
+}
+
+static struct aws_channel_handler *s_tls_handler_new_win10_plus(
+    struct aws_allocator *alloc,
+    struct aws_tls_connection_options *options,
+    struct aws_channel_slot *slot,
+    bool is_client_mode) {
+    AWS_ASSERT(options->ctx);
+
+    struct secure_channel_handler *sc_handler = aws_mem_calloc(alloc, 1, sizeof(struct secure_channel_handler));
+    if (!sc_handler) {
+        return NULL;
+    }
+    struct secure_channel_ctx *sc_ctx = options->ctx->impl;
+
+    SCH_CREDENTIALS credentials = {0};
+
+    ZeroMemory(&credentials, sizeof(SCH_CREDENTIALS));
+
+    TLS_PARAMETERS tls_params = {0};
+    if (sc_ctx->schannel_creds.enabledProtocols == 0) {
+        tls_params.grbitDisabledProtocols = 0;
+    } else {
+        tls_params.grbitDisabledProtocols = ~(sc_ctx->schannel_creds.enabledProtocols);
+    }
+    credentials.pTlsParameters = &tls_params;
+    credentials.cTlsParameters = 1;
+    credentials.dwSessionLifespan = 0; /* default 10 hours */

```

Comment:
Is this a bound on 1.3 session resumption?  Hopefully its not a bound on connection lifetime.

Reply:
The number of milliseconds that Schannel keeps the session in its session cache. After this time has passed, any new connections between the client and the server require a new Schannel session. Set the value of this member to zero to use the default value of 36000000 milliseconds (ten hours).

Reply:
it is similar to what we were using before.. with schannel_cred